examples/llama3-vllm-gcp/llama3_vllm_gcp.py

```diff
@@ -0,0 +1,188 @@
+# # Run Llama 3 8B Model Inference with vLLM on GCP
+
+# This example demonstrates how to run a Llama 3 8B model from Hugging Face
+# with vLLM on GCP using Runhouse.
+#
+# Make sure to sign the waiver on the [Hugging Face model page](https://huggingface.co/meta-llama/Meta-Llama-3-8B-Instruct)
+# so that you can access it.
+#
+# ## Setup credentials and dependencies
+#
+# Optionally, set up a virtual environment:
+# ```shell
+# $ conda create -n llama3-rh python=3.9.15
+# $ conda activate llama3-rh
+# ```
+# Install the required dependencies:
+# ```shell
+# $ pip install "runhouse[gcp] asyncio"
+# ```
+#
+# We'll be launching a GCP instance via [SkyPilot](https://github.com/skypilot-org/skypilot), so we need to
+# make sure your credentials are set up. You may be prompted to pick a cloud project to use after running `gcloud init`.
+# If you don't have one ready yet, you can connect one later by listing your projects
+# with `gcloud projects list` and setting one with `gcloud config set project <PROJECT_ID>`.
+# ```shell
+# $ gcloud init
+# $ gcloud auth application-default login
+# $ sky check
+# ```
+# We'll be downloading the Llama 3 model from Hugging Face, so we need to set up our Hugging Face token:
+# ```shell
+# $ export HF_TOKEN=<your huggingface token>
+# ```
+#
+# ## Define a Llama 3 model class
+# We import `runhouse` and `asyncio` because that's all that's needed to run the script locally.
+# The actual vLLM imports can happen within the functions that will be sent to the Runhouse cluster via an env.
```

Comment:
minor point of clarification - i read this as the env is responsible for sending the functions to the cluster, rather than representing the environment on the cluster in which the function itself is served

Reply:
Updated a little - lmk if that's better?